server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1398,9 +1400,34 @@ private void leaveRepoLoop(String repository) {
     }
 
     private void finalizeSnapshotEntry(Snapshot snapshot, Metadata metadata, RepositoryData repositoryData) {
-        assert currentlyFinalizing.contains(snapshot.getRepository());
-        assert repositoryOperations.assertNotQueued(snapshot);
-        try {
+        threadPool.executor(ThreadPool.Names.SNAPSHOT).execute(new SnapshotFinalization(snapshot, metadata, repositoryData));
+    }
+
+    /**
+     * Implements the finalization process for a snapshot: does some preparatory calculations, builds a {@link SnapshotInfo} and a
+     * {@link FinalizeSnapshotContext}, calls {@link Repository#finalizeSnapshot} and handles the outcome by notifying waiting listeners
+     * and triggering the next snapshot-related activity.
```

Comment:
This comment is vague. What's the next snapshot-related activity? What's the preparatory calculations (maybe remove this if not important to know)?  There's no context about `FinalizeSnapshotContext` and why it's relevant to surface a mention on the interface here.

How do I use this thing and why?

Reply:
The comment is more to orient the reader about what the method does because it's a little long right now (will be shorter soon). It's an `AbstractRunnable` so the only way to use it is to run it, and why is because you are finalizing a snapshot?

The next snapshot-related activity is deliberately vague because it kinda could be anything that was blocked on this finalization. This includes completing another snapshot, starting a batch of deletes, running a cleanup, likely others too.